### ISSUE TITLE
Issues with merging data from multiple coverage databases ("cannot locate structure")

### ISSUE THREAD

**Issue by nthykier** [pjcj/Devel--Cover#150]
Hi,

I got a Perl project called [lintian]([source]), which I would like to do coverage reports on. 
- I tried to run each test case in parallel with its own coverage database (see [rationale] if you are curious)
- Then I created a report for the databases

At this point, I got a a number of warnings a la:

```
Devel::Cover: Warning: can't locate structure for subroutine in lib/Lintian/Collect/Changes.pm
Devel::Cover: Warning: can't locate structure for statement in lib/Lintian/Collect/Changes.pm
Devel::Cover: Warning: can't locate structure for branch in lib/Lintian/Collect/Changes.pm
Devel::Cover: Warning: can't locate structure for condition in lib/Lintian/Collect/Changes.pm
```
- The warnings seem to depend on which database is the "primary" database and what files were covered by that test case.
- If I re-arrange the command line, so I pick a test that loads _all_ files, then the warnings go away.

My guess is that cover does not properly merge the databases, when there is a lazy loading involved (there is some of that in this project).

[rationale]: The majority of the test suite is a full black box test of the program.  We got over 400 of these test case, which takes at the very least 10 seconds each under Devel::Cover.

**Comment by pjcj** [comment]
It's taken a while, but I have been able to reproduce this.  I've put the results at https://pjcj.net/cover/cover_db-linital-seq/coverage.html and the results when running in parallel at https://pjcj.net/cover/cover_db-linital-par/coverage.html.  I'm not quite sure why there are differences, but I don't think there are any problems with parallel runs.

However, the problems here and in #151 are real.  But they are difficult to come across in a small test suite.  I'd love to fix this but need to be able to reproduce it in something smaller.

In addition, the merging of the DBs gets slower the more DBs that have been merged.  I'd like to look into that too.

I'll leave this open, but realistically there's not much that I can do right now.
